src/App.tsx

```diff
@@ -15,71 +26,98 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export class App extends React.Component {
+  state = {
+    selectedGood: 'Jam',
+    selectedClass: 'has-background-success-light',
+    noOneSelected: 'No goods selected',
+    addButtonText: '+',
+    removeButtonText: '-',
+  };
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  handleAddFunction = (event: React.MouseEvent<HTMLElement>): void => {
+    const { target } = event;
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+    if ((target as HTMLButtonElement).type === 'button') {
+      if (event.currentTarget.lastChild !== null) {
+        this.setState({
+          selectedGood: (event.currentTarget.lastChild as HTMLTableCellElement)
+            .innerText,
+        });
+      }
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+      event.currentTarget.classList.add(this.state.selectedClass);
+      (target as HTMLButtonElement).innerText = this.state.removeButtonText;
```

Comment:
do not work with DOM properties.
render conditionally in JSX text content, classes and so on.
for managing classes you can use library [classnames](https://www.npmjs.com/package/classnames)